src/devices/m_bus.c

```diff
@@ -310,7 +330,7 @@ static data_t *append_val(data_t *data, enum UnitType unit_type, uint8_t value_t
         fprintf(stderr, "M-Bus: Program error, exp (%d) is out of bounds", exp);
         return data;
     }
-    float fvalue = val * pow10_table[exp];
+    double fvalue = val * pow10_table[exp];
```

Comment:
Is double really needed?

Reply:
Not fully sure. But I increased it because we have here 64bit values. So to keep precision.

Reply:
This does not do what one might think. The calculation ist float, the float is then stored as double.

Reply:
you are right, I forgot to change also pow10_table